File: 卡码/代码随想录/JS/二叉树/01-二叉树遍历方式/LeetCode_0429.js
//429. N 叉树的层序遍历
function Node(val,children) {
  this.val = val;
  this.children = children;
};

const levelOrder = function(root) {
  const res=new Array();
  const q=new Array();
  if(root) q.push(root);
  while(q.length>0){
    let size=q.length;
    let v=new Array();
    while(size-->0){
      const node=q.shift();
      v.push(node.val);
      for(let child of node.children) q.push(child);
    }
    res.push(v);
  }
  return res;
};

File: 卡码/代码随想录/C++/二叉树/01-二叉树遍历方式/LeetCode_0429.cpp
//429. N ²æÊ÷µÄ²ãĞò±éÀú
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

class Node {
public:
  int val;
  vector<Node*> children;
  Node() {}
  Node(int _val) {
    val = _val;
  }
  Node(int _val, vector<Node*> _children) {
    val = _val;
    children = _children;
  }
};

class Solution {
public:
  vector<vector<int>> levelOrder(Node* root) {
    vector<vector<int>> res;
    vector<int> v;
    queue<Node*> q;
    int size;
    if(root) q.push(root);
    while(!q.empty()){
			size=q.size();
			v.clear();
			while(size-->0){
				Node* node=q.front();
				q.pop();
				v.push_back(node->val);
				for(Node* tmp:node->children){
					q.push(tmp);
				}
			}
			res.push_back(v);
		}
		return res;
  }
};

int main(){
  system("pause");
  return 0;
}
